File: Chess/ChessProblem_Test.cs
using System;
using System.IO;
using NUnit.Framework;

namespace Chess
{
	[TestFixture]
	public class ChessProblem_Test
	{
		private static void TestOnFile(string filename)
		{
			var board = File.ReadAllLines(filename);
			ChessProblem.LoadFrom(board);
			var expectedAnswer = File.ReadAllText(Path.ChangeExtension(filename, ".ans")).Trim();
			ChessProblem.CalculateChessStatus();
			Assert.AreEqual(expectedAnswer, ChessProblem.ChessStatus.ToString().ToLower(), "Failed test " + filename);
		}

		[Test]
		public void RepeatedMethodCallDoNotChangeBehaviour()
		{
			var board = new[]
			{
				"        ",
				"        ",
				"        ",
				"   q    ",
				"    K   ",
				" Q      ",
				"        ",
				"        ",
			};
			ChessProblem.LoadFrom(board);
			ChessProblem.CalculateChessStatus();
			Assert.AreEqual(ChessStatus.Check, ChessProblem.ChessStatus);
			
			// Now check that internal board modifictions during the first call do not change answer
			ChessProblem.CalculateChessStatus();
			Assert.AreEqual(ChessStatus.Check, ChessProblem.ChessStatus);
		}

		[Test]
		public void FullTests()
		{
			var dir = TestContext.CurrentContext.TestDirectory;
			var testsCount = 0;
			foreach (var filename in Directory.GetFiles(Path.Combine(dir, "ChessTests"), "*.in"))
			{
				TestOnFile(filename);
				testsCount++;
			}
			Console.WriteLine("Tests passed: " + testsCount);
		}
	}
}

File: java/src/main/java/ru/kontur/courses/chess/ChessProblemTest.java
package ru.kontur.courses.chess;

import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.FileInputStream;
import java.util.Arrays;
import java.util.Objects;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ChessProblemTest {

    @Test
    public void repeatedMethodCallDoNotChangeBehaviour() throws Exception {
        var boardLines = new String[]{
                "        ",
                "        ",
                "        ",
                "   q    ",
                "    K   ",
                " Q      ",
                "        ",
                "        ",
        };
        ChessProblem.loadFrom(boardLines);
        ChessProblem.calculateChessStatus();
        assertEquals(ChessStatus.CHECK, ChessProblem.chessStatus);

        // Now check that internal board modifications during the first call do not change answer
        ChessProblem.calculateChessStatus();
        assertEquals(ChessStatus.CHECK, ChessProblem.chessStatus);
    }

    @Test
    public void allTests() throws Exception {
        var resource = Thread.currentThread().getContextClassLoader().getResource("chessTests");
        assert resource != null;
        var dir = new File(resource.toURI());
        var inFiles = Arrays.stream(Objects.requireNonNull(dir.listFiles())).filter(it -> it.getName().contains(".in")).toList();
        for (var file : inFiles) {
            var in = new FileInputStream(file);
            var boardLines = new String(in.readAllBytes()).split("\n");
            in.close();
            ChessProblem.loadFrom(boardLines);
            var answerFile = new File(file.getAbsolutePath().replace(".in", ".ans"));
            var answerFileInput = new FileInputStream(answerFile);
            var answerText = new String(answerFileInput.readAllBytes()).trim();
            answerFileInput.close();

            ChessProblem.calculateChessStatus();
            assertEquals(answerText, ChessProblem.chessStatus.toString().toLowerCase());
        }
        System.out.println("Tests passed: " + inFiles.size());
    }
}
